include/fmt/base.h

```diff
@@ -1580,8 +1580,19 @@ template <> struct is_output_iterator<appender, char> : std::true_type {};
 
 template <typename It, typename T>
 struct is_output_iterator<
+#if FMT_GCC_VERSION >= 800 && FMT_GCC_VERSION <= 803
+    // Workaround old gcc bug, raised in issue 4129
+    It, T,
+    void_t<decltype(*std::declval<conditional_t<
+                        std::is_array<It>::value,
+                        typename std::add_pointer<
+                            typename std::remove_extent<It>::type>::type,
+                        It>&>()++ = std::declval<T>())>>
```

Comment:
> Can we just replace `It` with `std::decay<It>`?

Sure.
I omitted it and used the implementation of it in case of any edge case funniness (since it also does a bit more that _shouldn't_ matter but I was only 99.9% certain)